src/main.rs

```diff
@@ -38,19 +50,19 @@ fn main() -> Result<()> {
         )
     })?;
 
-    // new-config gets updated with selected values from old-config
-    new_config.update_with(&old_config);
-
     // new-config is written into conductor-config.toml file
     let new_config_toml = new_config
         .to_toml()
         .context("failed to serialize new_config to TOML")?;
-    std::fs::write(&config_path, &new_config_toml).with_context(|| {
-        format!(
-            "failed to write new_config to config_path ({})",
-            &config_path.display()
-        )
-    })?;
+    std::fs::write(&args.config_path, &new_config_toml).with_context(
+        || {
+            format!(
+                "failed to write new_config to config_path ({})",
+                &args.config_path.display()
+            )
+        },
+    )?;
+    utils::set_write_permissions(&args.config_path)?;
```

Comment:
This fails on HPOS with error:
```
Apr 16 15:08:39 hpos-1.holo.host systemd[1]: Starting holochain-conductor.service...
Apr 16 15:08:42 hpos-1.holo.host 36zhjz408pqch353sz110d9hcgpb5b96-unit-script-holochain-conductor-pre-start[3985]: Error: failed to set write permissions on /var/lib/holochain-conductor/conductor-config.toml
Apr 16 15:08:42 hpos-1.holo.host 36zhjz408pqch353sz110d9hcgpb5b96-unit-script-holochain-conductor-pre-start[3985]: Caused by:
Apr 16 15:08:42 hpos-1.holo.host 36zhjz408pqch353sz110d9hcgpb5b96-unit-script-holochain-conductor-pre-start[3985]:     Operation not permitted (os error 1)
```
It even fails when I `chmod 777 conductor-config.toml`. I think that it is an issue stemming from nixos that does not allow `pre-start` script to modify existing file. 

Also now I think that changing permissions from the level of the process is not the right thing to do, it should be taken care of in nixOS - most likely in the `pre-start` phase as well.

Reply:
Ok, I got it. Our permission problems stem from the fact that currently `holochain-conductor.service` [dynamically allocates process user](https://github.com/Holo-Host/holo-nixpkgs/blob/develop/modules/services/holochain-conductor.nix#L40). So obviously persistent files will cause permissions issue.

I suggest removing entire `set_write_permissions()` from this code and dealing with user on HPOS level - creating static user `holochain-conductor` and running systemd under this user.

Reply:
Thanks, I didn't think about it. Will address it in PR to `holo-nixpkgs`.

Reply:
Resolved in https://github.com/Holo-Host/holo-nixpkgs/pull/439/commits/30c323eb48a13fadf8d652d3aa3a38c1ab623477.